runtime/lua/vim/treesitter/highlighter.lua

```diff
@@ -312,23 +312,32 @@ function TSHighlighter._on_spell_nav(_, _, buf, srow, _, erow, _)
   end
 end
 
+---@private
+---@param buf integer
+function TSHighlighter._on_buf(_, buf)
+  local self = TSHighlighter.active[buf]
+  if self then
+    self.tree:parse(true)
```

Comment:
Isn't this the exact reason why this was changed?

`parse(true)` means the entire buffer is re-parsed every time any changes occur, which was not needed

Reply:
The problem is that it doesn't work. The range intersection code is extremely inefficient (worse than re-parsing) and freezes the editor on <100 LOC. See linked issue.

Reply:
That definitely isn't true. 

Try it on https://raw.githubusercontent.com/torvalds/linux/master/drivers/gpu/drm/amd/include/asic_reg/dcn/dcn_3_2_0_sh_mask.h